### DISCUSSION TITLE
Is there anyway to hide waybar?

### DISCUSSION THREAD

**Discussion by VicWang17** [prasanthrangan/hyprdots#1853] [Q&A] [answered]
I want to hide waybar sometimes, so I use this code after searching on Google:
```
bind = $mainMod, B, exec, killall -SIGUSR1 waybar
```
The problem is all the UI is hidden successfully but there is still a translucent bar on it.

So is there anyway to hide waybar or the translucent bar also can be hidden?

**Comment by rubiin** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/7174b07a-c95d-4fc3-b4aa-b3bc1c2f81e9)

**Comment by kRHYME7** [comment]
+  That's not waybar, but the layer hyprland gives for waybar. The moment you hide waybar  (-SIGUSR1), hyprland don't have any idea

+  You might want to ask for this feature in waybar repo. 


+ For now I can only give you a command to enable and to disable

**enable**

``` hyprctl --batch "keyword layerrule unset,waybar ; keyword layerrule blur,waybar" ```

**disable**

``` hyprctl --batch "keyword layerrule unset,waybar ; keyword layerrule ignorezero,waybar ``` 

**toggle**

Now I can't see any options to query the state of waybar (showing or not ) as hyprland always thinks that it's displayed somehow (reserved).
We can store the "state" (showing or not ) in a file, but this is unreliable.

You can: 
+ request waybar to handle the waybar layer on SIGUSR1 command 
+ (simple) request to waybar a query state so users can check it 
+ request hyprland to have an option to query the layers

**Reply by VicWang17** [reply]
Thx
